src/main/java/seedu/address/logic/commands/EditMeetingCommand.java

```diff
@@ -0,0 +1,215 @@
+import seedu.address.model.Model;
+import seedu.address.model.meeting.Description;
+import seedu.address.model.meeting.Meeting;
+import seedu.address.model.meeting.MeetingDate;
+import seedu.address.model.meeting.MeetingTime;
+
+
+/**
+ * Edits the details of an existing meeting in MyInsuRec.
+ */
+public class EditMeetingCommand extends Command {
+
+    public static final String COMMAND_WORD = "editMeeting";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the meeting identified "
+            + "by the index number used in the displayed meeting list. \n"
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) \n"
+            + "[" + PREFIX_DATE + "DATE] "
+            + "[" + PREFIX_START_TIME + "START TIME] "
+            + "[" + PREFIX_END_TIME + "END TIME] "
+            + "[" + PREFIX_DESCRIPTION + "DESCRIPTION]\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_INDEX + "1 "
+            + PREFIX_DATE + "23122022 ";
+
+    public static final String MESSAGE_EDIT_MEETING_SUCCESS = "Edited Meeting: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_MEETING = "This meeting already exists in MyInsuRec.";
+
+    private final Index index;
+    private final EditMeetingDescriptor editMeetingDescriptor;
+
+    /**
+     * @param index of the meeting in the filtered meeting list to edit
+     * @param editMeetingDescriptor details to edit the meeting with
+     */
+    public EditMeetingCommand(Index index, EditMeetingDescriptor editMeetingDescriptor) {
+        requireNonNull(index);
+        requireNonNull(editMeetingDescriptor);
+
+        this.index = index;
+        this.editMeetingDescriptor = new EditMeetingDescriptor(editMeetingDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Meeting> lastShownList = model.getFilteredMeetingList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_MEETING_DISPLAYED_INDEX);
+        }
+
+        Meeting meetingToEdit = lastShownList.get(index.getZeroBased());
+        Meeting editedMeeting = createEditedMeeting(meetingToEdit, editMeetingDescriptor);
+
+        if (!meetingToEdit.equals(editedMeeting) && model.hasSpecificMeeting(editedMeeting)) {
+            throw new CommandException(MESSAGE_DUPLICATE_MEETING);
+        }
+
+        // update meeting list
+        model.setMeeting(meetingToEdit, editedMeeting);
+        model.updateFilteredMeetingList(PREDICATE_SHOW_ALL_MEETING);
+
+        // update client list
+        model.setClient(meetingToEdit.getClient(), editedMeeting.getClient());
+        model.updateFilteredClientList(PREDICATE_SHOW_ALL_CLIENTS);
```

Comment:
understand this helps to update the ui of client, but not sure if it is necessary here. calling listClient also performs an update already.